shared/ssl.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/sh
+#
+# Get a signed certificate for this host.
+#
+# Uses OpenSSL and curl directly to generate a new CSR and get it signed by the
+# Puppet Server CA.
+#
+# Intended to be used in place of a full-blown puppet agent run that is solely
+# for getting SSL certificates onto the host.
+#
+# Files will be placed in the same default directory location and structure
+# that the puppet agent would put them, which is /etc/puppetlabs/puppet/ssl.
+#
+# Fails under the following conditions:
+# - CA host cannot be reached
+# - CA already has signed certificate for this host
+# - SSL files already exist on this host
+#
+# Arguments:
+#   $1  (Optional) Certname to use, defaults to $HOSTNAME
+#
+# Optional environment variables:
+#   WAITFORCERT            Number of seconds to wait for certificate to be
+#                          signed, defaults to 120
+#   PUPPETSERVER_HOSTNAME  Hostname of Puppet Server CA, defaults to "puppet"
+#   SSLDIR                 Root directory to write files to, defaults to
+#                          "/etc/puppetlabs/puppet/ssl"
+
+msg() {
+    echo "($0) $1"
+}
+
+error() {
+    msg "Error: $1"
+    exit 1
+}
+
+### Verify dependencies available
+! type -p openssl > /dev/null && error "openssl not found on PATH"
+! type -p curl > /dev/null && error "curl not found on PATH"
+
+### Verify options are valid
+CERTNAME="${1:-${HOSTNAME}}"
```

Comment:
puppet's certname is by default [lowercased hostname](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/ssl/host.rb#L100). I don't know if you want to be bug-for-bug compatible though?

Reply:
Haha great info, thanks Josh.

We explicitly set the certname via `puppet config set certname ...` in the puppetdb and puppetserver containers, so we wouldn't be relying on defaults there at least.

I'm inclined to not match it bug-for-bug at this point, unless it becomes clear that'd actually be better behavior.